doc/en/api/1.0.0/styles.yaml

```diff
@@ -660,33 +640,81 @@ paths:
 definitions:
 
   StyleInfoPost:
-    title: style
-    type: object
+    name: styleBody
     description: >
-    
-      The style body of a request.
-      JSON or XML style uploads follow the schema below.
+
+      The style body of a request
+      JSON or XML style uploads follow this schema. For example:
+
+        - application/xml: 
+          
+          ```
+          <style><name>roads_style</name><filename>roads.sld</filename></style>
+          ```
```

Comment:
These are both complete examples, rather than partial data structure to update ...

Examples to consider:
- minimal example with name and filename; it is complete enough for a POST? What format would it assume ...
- complete example with name, filename, format, legend, ...

What was our approach going to be?
- POST: create using a minimal viable example
- PUT: update using a minimal example
- GET: providing a maximal example showing all the stuff, even those things we cannot edit ...

Reply:
_Technically_, a minimal PUT is just an empty object; the minimum useful PUT is any one attribute, so that is still not all that helpful.

I do like using GET for the maximal example, although there may be attributes you can GET that you can't PUT, depending on the object in question. I'm not entirely certain about this though.

Reply:
Well a minimal put with an empty object would not do anything right?

If you wanted to we could do a maximal PUT example that shows all the editable things; there is a good chance we could use the same example for GET and PUT for the majority of end points.

How about:

* POST: example create using a minimal viable example
* PUT: example showing all editable information
* GET: Only provide an example if required to show extra non-editable information

Reply:
I like this approach, sounds good. 
Although in the cases where GET and PUT are the same, we may still want to ref the PUT example from the GET, since they do show up under seperate headings, and some people may only look at one or the other.

Reply:
yes, well played and good reduction of duplication